server/src/server_browser_backend/server_list.py

```
@@ -50,6 +50,21 @@ def update(
         self.servers[server_id] = result
 
         return result.resource
+    
+    def delete(
+        self, server_id: str, key: str
+    ) -> Server:
+        """Remove the server with the given id from the server list and return it."""
+        server = self.servers[server_id]
+        if server is None:
+            return None
```

This will throw a key error rather than returning None if something is missing.  This should be 
```python
server = self.servers.get(server_id)
```

This means some other functions in here are also wrong